mcs/class/System.Core/System.IO.Pipes/UnixDomainSocketEndPoint.cs

```diff
@@ -0,0 +1,140 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// Taken from CoreFX, merges UnixDomainSocketEndPoint.cs and UnixDomainSocketEndPoint.Unix.cs
+// and makes the class internal.
+
```

Comment:
Why cannot we use the original files instead?

Reply:
We can, but it will expose the UnixDomainSocketEndPoint as public and from wrong assembly. Moreover it's used only on Unix platforms and not on Windows and mobile profiles, so the other variants would have to be brought in. The type is not present on NetFX, but logically it is part of System.Net.Sockets and thus should be somewhere close-by.

Reply:
the type should really be in System.dll but it's not yet used by anything else for now. 

Could you please change the visibility of the type in Mono CoreFX fork using

e.g.
```
#if !MONO
public
#endif
```

and include these 2 files directly